src/main/java/uk/gov/companieshouse/company/profile/service/CompanyProfileService.java

```diff
@@ -95,7 +95,13 @@ public void updateInsolvencyLink(String contextId, String companyNumber,
 
         cpDocument.setCompanyProfile(companyProfileRequest.getData());
 
-        cpDocument.getUpdated().setAt(LocalDateTime.now().truncatedTo(ChronoUnit.SECONDS));
+        if (cpDocument.getUpdated() != null) {
+            cpDocument.getUpdated().setAt(LocalDateTime.now().truncatedTo(ChronoUnit.SECONDS));
```

Comment:
It should be LocalDateTime in UTC instead of server's current time, we should review all places we using now() or currentTime in our source codes